compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java

```diff
@@ -281,8 +287,13 @@ public void visit(BStructureType bStructureType) {
     @Override
     public void visit(BTupleType bTupleType) {
         buff.writeInt(bTupleType.tupleTypes.size());
-        for (BType memberType : bTupleType.tupleTypes) {
-            writeTypeCpIndex(memberType);
+        int i = 0;
+        for (BTupleMemberType memberType : bTupleType.tupleTypes) {
```

Comment:
Why can't we just use an indexed for loop?

Reply:
changed to an indexed loop.